apps/color-picker/src/locations/ConfigScreen.tsx

```diff
@@ -0,0 +1,254 @@
+/** @jsxImportSource @emotion/react */
+
+import { AppExtensionSDK } from '@contentful/app-sdk';
+import {
+  Button,
+  Card,
+  Flex,
+  Note,
+  Paragraph,
+  Subheading,
+  TextLink,
+} from '@contentful/f36-components';
+import { ExternalLinkIcon, PlusIcon } from '@contentful/f36-icons';
+import tokens from '@contentful/f36-tokens';
+import { useSDK } from '@contentful/react-apps-toolkit';
+import { css, Global } from '@emotion/react';
+import { useCallback, useEffect, useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import SwatchEditor from '../components/SwatchEditor';
+
+export interface AppInstallationParameters {
+  themes: Theme[];
+}
+
+const ConfigScreen = () => {
+  const [isInstalled, setIsInstalled] = useState(false);
+  const [parameters, setParameters] = useState<AppInstallationParameters>({
+    themes: [
+      {
+        id: 'default',
+        name: 'Default',
+        colors: [],
+      },
+    ],
+  });
+  const sdk = useSDK<AppExtensionSDK>();
+
+  const addSwatch = () => {
+    setParameters({
+      ...parameters,
+      themes: [
+        {
+          ...parameters.themes[0],
+          colors: [
+            ...parameters.themes[0].colors,
+            { id: uuidv4(), name: '', value: '#0088cc' },
+          ],
+        },
+      ],
+    });
```

Comment:
What about extracting the "swatch part" as a new state and then only on saving add it to parameters? Then you don't need to do this spreading all the time?